lib/ruby_lsp/document.rb

```diff
@@ -30,6 +30,7 @@ def initialize(source:, version:, uri:, encoding: Constant::PositionEncodingKind
       @uri = T.let(uri, URI::Generic)
       @unparsed_edits = T.let([], T::Array[EditShape])
       @parse_result = T.let(YARP.parse(@source), YARP::ParseResult)
+      @parse_result.attach_comments!
```

Comment:
@vinistock do you think we should always attach the comments, or make it conditional for requests that need it?

Reply:
It could indeed have a performance cost because it basically analyzes the entire AST to attach the comments.

Given that the only request that needs comments is document link, which we plan on removing in the future anyway, I'm not sure we want to do it.

Tapioca always adds the magic source comment immediately above the class/module/constant/method, so we always know where to find it without paying this cost.

What do you think about avoiding it for now to get the most performance we can out of YARP?

In that case, we can pass the comments array to document link's initializer, turn the array into a hash of `line_number => comment` and then just access it directly when finding nodes.

Reply:
I'll push a commit to show how that would look, then we can decide.

Reply:
Pushed.